src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerServiceUpdateTest.java

```diff
@@ -129,7 +134,46 @@ void shouldFailUpdateIfNationalitiesAreSame() {
         );
 
         assertEquals("Second nationality must be different from the first", Objects.requireNonNull(exception.getBindingResult().getFieldError("nationality2")).getDefaultMessage());
+    }
+
+    @Test
+    void shouldAllowUpdateIfNationalitiesAreDifferent() throws Exception {
+        GeneralPartnerDao generalPartnerDao = createGeneralPartnerPersonDao();
+
+        GeneralPartnerDataDto generalPartnerDataDto = new GeneralPartnerDataDto();
+        generalPartnerDataDto.setNationality1(Nationality.AMERICAN);
+        generalPartnerDataDto.setNationality2(Nationality.NEW_ZEALANDER);
+
+        when(repository.findById(generalPartnerDao.getId())).thenReturn(Optional.of(generalPartnerDao));
+
+        service.updateGeneralPartner(GENERAL_PARTNER_ID, generalPartnerDataDto, REQUEST_ID, USER_ID);
+
+        verify(repository).save(submissionCaptor.capture());
+
+        GeneralPartnerDao sentSubmission = submissionCaptor.getValue();
+
+        assertEquals(Nationality.AMERICAN.getDescription(), sentSubmission.getData().getNationality1());
+        assertEquals(Nationality.NEW_ZEALANDER.getDescription(), sentSubmission.getData().getNationality2());
+    }
+
+    @Test
```

Comment:
what happens in the scenario where they had 2 nationalities, but they set both to null, should that fail on nationality1 being null for a mandatory field?

Reply:
No, it won't fail on the API side because this an update (PATCH) call and any of the fields can be null if not sent - so no mandatory checks. It's just that in the case of the second nationality being null but a first nationality isn't null (i.e. user has submitted data on the 'Add General Partner Person' page and issued an 'update' not 'create' call to the API) this is a clear case that the second nationality should be cleared/reset.